spec/cookie_spec.rb

```diff
@@ -0,0 +1,167 @@
+        end
+      end
+    end
+
+    context 'when the cookie and Authorization header are both missing' do
+      it 'decodes the cookie and adds the token to the headers' do
+        get('/')
+
+        aggregate_failures do
+          expect(last_response.status).to eq 401
+          expect(parsed_body).to eq(error: 'Missing token cookie and Authorization header')
+          expect(last_response.headers['jwt.header']).to be_nil
+        end
+      end
+    end
+
+    context 'when a valid Authorization header, but no cookie is sent with the request' do
+      it 'decodes token' do
+        header 'Authorization', "Bearer #{Rack::JWT::Token.encode(token_payload, secret, 'HS256')}"
+        get('/')
+
+        aggregate_failures do
+          expect(last_response.status).to eq 200
+          expect(parsed_body).to eq(token_payload)
+          expect(last_response.headers['jwt.header']).to eq('typ' => 'JWT', 'alg' => 'HS256')
+        end
+      end
+    end
+
+    context 'when a valid Authorization header, and a valid auth cookie are both sent with the request' do
+      it 'decodes token' do
+        header 'Authorization', "Bearer #{Rack::JWT::Token.encode(token_payload, secret, 'HS256')}"
+        set_cookie("authToken=#{token}")
+        get('/')
+
+        aggregate_failures do
+          expect(last_response.status).to eq 200
+          expect(parsed_body).to eq(token_payload)
+          expect(last_response.headers['jwt.header']).to eq('typ' => 'JWT', 'alg' => 'HS256')
+        end
+      end
+    end
+
+    context 'when an empty cookie is sent with the request' do
+      it 'decodes the cookie and adds the token to the headers' do
+        set_cookie('authToken=')
+        get('/')
+
+        aggregate_failures do
+          expect(last_response.status).to eq 401
+          expect(parsed_body).to eq(error: 'Empty token cookie')
+          expect(last_response.headers['jwt.header']).to be_nil
+        end
+      end
+    end
+  end
+
+  context 'when the cookie setting is disabled' do
+    let(:decode_args) do
+      {
+        secret: secret,
+        verify: true,
+        options: {
+          algorithm: 'HS256',
+          verify_expiration: true,
+          verify_not_before: true,
+          iss: 'https://my.awesome.website/',
+          verify_iss: true,
+          verify_iat: true,
+          jti: jti_digest,
+          verify_jti: true,
+          aud: 'audience',
+          verify_aud: true,
+          sub: 'subject',
+          verify_sub: true,
+          leeway: 30
+        }
+      }
+    end
+
+    context 'when the cookie and Authorization header are both missing' do
+      it 'decodes the cookie and adds the token to the headers' do
+        get('/')
+
+        aggregate_failures do
+          expect(last_response.status).to eq 401
+          expect(parsed_body).to eq(error: 'Missing Authorization header')
+          expect(last_response.headers['jwt.header']).to be_nil
+        end
+      end
+    end
+  end
+end
```

Comment:
Add spec within `cookie is disabled` to decode the token from the header successfully?